sql/plan/merge_join.go

```diff
@@ -84,20 +85,29 @@ func newMergeJoinIter(ctx *sql.Context, j *JoinNode, row sql.Row) (sql.RowIter,
 // not provide a directional ordering signal for index iteration
 // are evaluated separately.
 type mergeJoinIter struct {
-	cmp     expression.Comparer
+	// cmp is a directional indicator for row iter increments
+	cmp expression.Comparer
+	// filters is the remaining set of join conditions
 	filters []sql.Expression
 	left    sql.RowIter
 	right   sql.RowIter
 	fullRow sql.Row
 
-	// match lookahead buffers and state tracking
+	// match lookahead buffers and state tracking (private to match)
 	rightBuf  []sql.Row
 	bufI      int
 	rightPeek sql.Row
 	leftPeek  sql.Row
 	rightDone bool
 	leftDone  bool
 
+	// matchIncLeft indicates whether the most recent |i.incMatch|
+	// call incremented the left row.
+	matchIncLeft bool
+	// leftMatched indicates whether the current left in |i.fullRow|
+	// has satisfied the join condition.
+	leftMatched bool
```

Comment:
These docs are also super helpful. They do a good job encoding the intent of how these are supposed to be used, so it's crystal clear and a future reader/editor doesn't have to make any guesses.